### ISSUE TITLE
[bug] `no-nested-lazy-component-declarations` should not be reported when it depends on internal variables and already wrapped in `useMemo`

### ISSUE THREAD

**Issue by JounQin** [Rel1cx/eslint-react#1089]
### Describe the bug

```tsx
const Readme = () => {
  const { name } = useParams<'name'>()
  const Readme = useMemo(
    () =>
      // eslint-disable-next-line @eslint-react/no-nested-lazy-component-declarations
      lazy(() =>
        name ? import(`../packages/${name}/README.md`) : import('../README.md'),
      ),
    [name],
  )
  return (
    <Suspense>
      <Readme />
    </Suspense>
  )
}
```

I don't think it's possible to externalize `lazy()`

### Reproduction

https://github.com/un-ts/mono-lib-boilerplate/blob/ec53a6c51dc61ade5efad488f3b4aba07568d5dc/docs/App.tsx#L19

### Expected behavior

No error

### Platform and versions

```Text
System:
    OS: macOS 15.4.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 1.75 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.14.0 - ~/.local/state/fnm_multishells/86403_1744990861557/bin/node
    Yarn: 4.9.1 - ~/.local/state/fnm_multishells/86403_1744990861557/bin/yarn
    npm: 9.9.4 - ~/.local/state/fnm_multishells/86403_1744990861557/bin/npm
    bun: 1.2.9 - /opt/homebrew/bin/bun
  Browsers:
    Chrome: 135.0.7049.96
    Safari: 18.4
  npmPackages:
    @eslint-react/eslint-plugin: ^1.48.4 => 1.48.4
```

### Stack trace

```Text

```

**Comment by Rel1cx** [comment]
The code above has already required three `eslint-disable-next-line` disable comments to achieve what it's trying to do. I think is quite reasonable that one of the them is for our rule.

**Comment by JounQin** [comment]
> The code above has already required three `eslint-disable-next-line` disable comments to achieve what it's trying to do. I think is quite reasonable that one of the them is for our rule.

The codes can be simplified without disabling other rules...

---

@Rel1cx Example updated.

**Comment by Rel1cx** [comment]
> > The code above has already required three `eslint-disable-next-line` disable comments to achieve what it's trying to do. I think is quite reasonable that one of the them is for our rule.
> 
> The codes can be simplified without disabling other rules...
> 
> [@Rel1cx](https://github.com/Rel1cx) Example updated.

Good, I personally think this usage forces the client side `react-router` to do _meta-framework like dynamic routing_ (with the least suitable declarative mode of the [three modes](https://reactrouter.com/start/modes#picking-a-mode)), so it makes sense to require a disable comment.

But I'm going to leave this issue for a little bit longer for other people to give their opinion on this example.

**Comment by Rel1cx** [comment]
https://github.com/jacob-ebey/react-router-lazy-mdx/blob/main/app/routes/post.tsx This is similar to your example. There is no `lazy()` that needs to be used.

**Comment by JounQin** [comment]
@Rel1cx 

That requires:

1. `import.meta.glob`
2. `use(Promise)`
